packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -2463,10 +2477,17 @@ function flushPassiveEffectsImpl() {
 
   flushSyncCallbacks();
 
-  // If additional passive effects were scheduled, increment a counter. If this
-  // exceeds the limit, we'll fire a warning.
-  nestedPassiveUpdateCount =
-    rootWithPendingPassiveEffects === null ? 0 : nestedPassiveUpdateCount + 1;
+  if (__DEV__) {
+    // If additional passive effects were scheduled, increment a counter. If this
+    // exceeds the limit, we'll fire a warning.
+    if (didScheduleUpdateDuringPassiveEffects) {
```

Comment:
Will this work if you have 50+ separate roots that update simultaneously? I remember we got a bug report for that once, where each root was a separate tooltip on a Google Maps embed, or something. I think that was for the sync version of this guard, though, and it's kind of an edge case.

Reply:
Pushed a fix, same as we had for the sync case.